### ISSUE TITLE
lightgbm categorical issue

### ISSUE THREAD

**Issue by nirupamkar** [marcotcr/lime#290]
I am trying to replicate the below code written for random forest to do the same for lightgbm with categorical variables.
https://github.com/marcotcr/lime/blob/master/doc/notebooks/Tutorial%20-%20continuous%20and%20categorical%20features.ipynb

Only difference is that the lightgbm model does not use onehot encoding. I want to use the variable as raw format. Please find below the code

`import lightgbm as lgb
import pandas as pd
import sklearn.ensemble
import numpy as np
import lime
import lime.lime_tabular
data = np.genfromtxt('/agaricus-lepiota.data', delimiter=',', dtype='<U20')
labels = data[:,0]
le= sklearn.preprocessing.LabelEncoder()
le.fit(labels)
labels = le.transform(labels)
class_names = le.classes_
data = data[:,1:]

feature_names = 'cap-shape,cap-surface,cap-color,bruises?,odor,gill-attachment,gill-spacing,gill-size,gill-color,stalk-shape,stalk-root,stalk-surface-above-ring, stalk-surface-below-ring, stalk-color-above-ring,stalk-color-below-ring,veil-type,veil-color,ring-number,ring-type,spore-print-color,population,habitat'.split(',')
data=pd.DataFrame(data)
data.columns=feature_names
categorical_features=feature_names

categorical_names = '''bell=b,conical=c,convex=x,flat=f,knobbed=k,sunken=s
fibrous=f,grooves=g,scaly=y,smooth=s
brown=n,buff=b,cinnamon=c,gray=g,green=r,pink=p,purple=u,red=e,white=w,yellow=y
bruises=t,no=f
almond=a,anise=l,creosote=c,fishy=y,foul=f,musty=m,none=n,pungent=p,spicy=s
attached=a,descending=d,free=f,notched=n
close=c,crowded=w,distant=d
broad=b,narrow=n
black=k,brown=n,buff=b,chocolate=h,gray=g,green=r,orange=o,pink=p,purple=u,red=e,white=w,yellow=y
enlarging=e,tapering=t
bulbous=b,club=c,cup=u,equal=e,rhizomorphs=z,rooted=r,missing=?
fibrous=f,scaly=y,silky=k,smooth=s
fibrous=f,scaly=y,silky=k,smooth=s
brown=n,buff=b,cinnamon=c,gray=g,orange=o,pink=p,red=e,white=w,yellow=y
brown=n,buff=b,cinnamon=c,gray=g,orange=o,pink=p,red=e,white=w,yellow=y
partial=p,universal=u
brown=n,orange=o,white=w,yellow=y
none=n,one=o,two=t
cobwebby=c,evanescent=e,flaring=f,large=l,none=n,pendant=p,sheathing=s,zone=z
black=k,brown=n,buff=b,chocolate=h,green=r,orange=o,purple=u,white=w,yellow=y
abundant=a,clustered=c,numerous=n,scattered=s,several=v,solitary=y
grasses=g,leaves=l,meadows=m,paths=p,urban=u,waste=w,woods=d'''.split('\n')
for j, names in enumerate(categorical_names):
    values = names.split(',')
    values = dict([(x.split('=')[1], x.split('=')[0]) for x in values])
    data[feature_names[j]] = np.array(list(map(lambda x: values[x], data[feature_names[j]])))

categorical_names = {}
for feature in categorical_features:
    le = sklearn.preprocessing.LabelEncoder()
    le.fit(data[ feature])
    data[ feature] = le.transform(data[ feature])
    categorical_names[feature] = le.classes_
    data[feature]=data[feature].astype("category")

train, test, labels_train, labels_test = sklearn.model_selection.train_test_split(data, labels, train_size=0.80)

rf = lgb.LGBMClassifier()
rf.fit(train, labels_train,categorical_feature=categorical_features)

predict_fn = lambda x: rf.predict_proba((x))

explainer = lime.lime_tabular.LimeTabularExplainer(train.values ,class_names=['edible', 'poisonous'], feature_names = feature_names,
                                                   categorical_features=categorical_features, 
                                                   categorical_names=categorical_names, kernel_width=3, verbose=False)

i = 6734
#might need to change this number
exp = explainer.explain_instance(test.ix[i].values, predict_fn, num_features=5)
exp.show_in_notebook()`

Everything ran without an error. But in the graph, it seems it's treating the features as numeric.
Any help on this?
<img width="971" alt="screenshot 2019-02-06 at 9 37 09 pm" src="https://user-images.githubusercontent.com/37179907/52355023-8990cb80-2a57-11e9-9c6e-01951ff806c7.png">

**Comment by marcotcr** [comment]
`categorical_features` should be a list of indices (integers) rather than strings. You could do ` categorical_features=list(range(len(feature_names)))`
